src/OmniSharp.MSBuild/MSBuildProjectSystem.cs

```diff
@@ -400,6 +406,31 @@ private void UpdateSourceFiles(Project project, IList<string> sourceFiles)
             }
         }
 
+        private void WatchDirectoryContainingFile(string sourceFile) => _fileSystemWatcher.WatchDirectory(Path.GetDirectoryName(sourceFile), OnDirectoryFileChanged);
+
+        private void OnDirectoryFileChanged(string path, FileChangeType? changeType)
+        {
+            // Hosts may not have passed through a file change type
+            if (changeType == null && !File.Exists(path) || changeType.GetValueOrDefault() == FileChangeType.Delete)
```

Comment:
Minor nit it's more obvious that the value is Nullable<> if we use `!changeType.HasValue`.